statsmodels/stats/multivariate.py

```diff
@@ -59,6 +59,41 @@ def test_mvmean(data, mean_null=0, return_results=True):
     else:
         return statistic, pvalue
 
+def test_mvmean_two_sample(x, y, return_results=True):
+    """Hotellings test for multivariate mean in two samples
+
+    Parameters
+    ----------
+    x : array_like
+        first sample data with observations in rows and variables in columns
+    y : array_like
+        second sample data with observations in rows and variables in columns
+
+    Returns
+    -------
+    results : instance of a results class with attributes
+        statistic, pvalue, t2 and df
+    """
+    x = array_like(x, "x", ndim=2)
+    y = array_like(y, "x", ndim=2)
+    nobs_x, k_vars = x.shape
+    nobs_y, k_vars = y.shape
+    mean_x = x.mean(0)
+    mean_y = y.mean(0)
+    cov_x = np.cov(x, rowvar=False, ddof=1)
+    cov_y = np.cov(y, rowvar=False, ddof=1)
+    combined_cov = ((nobs_x - 1) * cov_x + (nobs_y - 1) * cov_y) / (nobs_x - 1 + nobs_y - 1)
+    diff = mean_x - mean_y
+    t2 = (nobs_x * nobs_y) / (nobs_x + nobs_y) * diff @ (np.linalg.solve(combined_cov, diff))
+    factor = ((nobs_x + nobs_y - 2) * k_vars) / (nobs_x + nobs_y - k_vars - 1)
+    statistic = t2 / factor
+    df = (k_vars, nobs_x + nobs_y - 1 - k_vars)
+    pvalue = stats.f.sf(statistic, df[0], df[1])
+    return HolderTuple(statistic=statistic,
```

Comment:
missing `    if return_results:` for conditional return given the keyword option

Reply:
think @bashtage wanted to remove the keyword option and always return the full result set.

Reply:
that's fine also, either way, I'm still ambiguous about this as long as results instance is default.

I still prefer having the no-results option for functions where I think I might use bootstrap, simulations or compute it over a large set of cases.